src/evict/evict_lru.c

```diff
@@ -1617,6 +1620,22 @@ __evict_walk(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue)
         goto retry;
     }
 
+#ifdef HAVE_DIAGNOSTIC
+    if (!F_ISSET(conn, WT_CONN_IN_MEMORY) && cache->evict_aggressive_score == WT_EVICT_SCORE_MAX &&
+      (slot == queue->evict_entries || slot > start_slot)) {
```

Comment:
Just a question, what is this check for?

Reply:
This is to check we don't add enough pages to the queue.